src/app/views/sessions/signin/store/signin.store.ts

```diff
@@ -191,7 +192,8 @@ export class SigninStore extends ComponentStore<SigninState> {
   }
 
   private authenticateWithToken(): Observable<unknown> {
-    return this.ws.call('auth.generate_token', [this.tokenLifetime]).pipe(
+    const tokenLifetime = Number(this.window.localStorage.getItem('lifetime')) || this.tokenLifetimeDefault;
+    return this.ws.call('auth.generate_token', [tokenLifetime]).pipe(
```

Comment:
I think `tokenLifetime` doesn't matter too much here, as AFAIK any message, including pings will extend its duration.
It's worth checking if this is true and if true, remove localstorage part from here.